db/eventlog.c

```diff
@@ -72,11 +73,19 @@ void eventlog_init(const char *dbname)
     if (eventlog_enabled) eventlog = eventlog_open();
 }
 
+static inline void free_gbl_eventlog_fname()
+{
+    if (gbl_eventlog_fname == NULL)
+        return;
+    free(gbl_eventlog_fname);
+    gbl_eventlog_fname = NULL;
+}
+
 static gzFile eventlog_open()
 {
     char *fname = eventlog_fname(thedb->envname);
+    gbl_eventlog_fname = fname;
     gzFile f = gzopen(fname, "2w");
-    free(fname);
     if (f == NULL) {
         eventlog_enabled = 0;
```

Comment:
free_gbl_eventlog_fname() will never be called, as eventlog will be null in this case.

Reply:
Good catch. Not sure under what condition (and how often) could gzopen() fail, but fixing so when it does, we free.
Thanks Mohit.